manageability-and-operations/observability-and-manageability/oci-monitoring/custom-metrics/custom-metric-FN-services-limit-monitoring/files/Function/func.py

```diff
@@ -0,0 +1,244 @@
+        )
+    else:
+        post_metric_data_response = monitoring_client.post_metric_data(
+            post_metric_data_details=oci.monitoring.models.PostMetricDataDetails(
+                metric_data=[
+                    oci.monitoring.models.MetricDataDetails(
+                        namespace = "limits_metrics",
+                        compartment_id = compartment_ocid,
+                        name = m_name,
+                        dimensions={
+                            'service_name': s_name,
+                            'limit_name': l_name,
+                            'availability_domain': a_domain
+                            },
+                        datapoints=[
+                            oci.monitoring.models.Datapoint(
+                                timestamp=datetime.datetime.strftime(
+                                    times_stamp,"%Y-%m-%dT%H:%M:%S.%fZ"),
+                                value = value)]
+                        )]
+            )
+        )
+    return post_metric_data_response
+
+# getServiceLimitsUsage: gets the existing limits for a service and limit name in a compartment and, if AD specific, for its AD
+def getServiceLimitsUsage(s_name, l_name, compartment_ocid, limits_client, a_domain=None):
+    if a_domain is None:
+        # We gather the service limit usage
+        get_resource_availability_response = limits_client.get_resource_availability(service_name = s_name, limit_name = l_name, compartment_id = compartment_ocid)
+
+        # We need to gather the service limit limit
+        list_limit_values_response = limits_client.list_limit_values(compartment_id = compartment_ocid, service_name = s_name, limit = 1)
+    else:
+        # We gather the service limit usage
+        get_resource_availability_response = limits_client.get_resource_availability(service_name = s_name, limit_name = l_name, compartment_id = compartment_ocid, availability_domain = a_domain)
+
+        # We need to gather the service limit limit
+        list_limit_values_response = limits_client.list_limit_values(compartment_id = compartment_ocid, service_name = s_name, availability_domain = a_domain, limit = 1)
+
+    usage = json.loads(str(get_resource_availability_response.data))
+    used = usage["used"]
+    available = usage["available"]
+    limit_limit = json.loads(str(list_limit_values_response.data[0]))
+    max_limit = limit_limit["value"]
+
+    # We create the return type data dictionary
+    values = {
+        "used": used,
+        "available": available,
+        "max_limit": max_limit
+    }
+    return json.dumps(values)
+
+def handler(ctx, data: io.BytesIO = None):
+
+    resp={}
+    resp["Result"] = "OK"
+
+    # We gather the input arguments from the function configuration:
+    try:
+       cfg = dict(ctx.Config())
+       compartment_ocid = cfg["compartment_ocid"]
+       region = cfg["region"]
+    except (Exception, ValueError) as ex:
+       logging.getLogger().error('ERROR: ' + str(ex))
```

Comment:
Do not get the logger every time. Get it once at the beginning and use the variable.

logger = logging.getLogger()
...
logger.error("Message.")

You don't have to include the text "ERROR:".